src/Utilities/set_eval.jl

```diff
@@ -0,0 +1,178 @@
+
+function _correct_distance_to_set(d::Real)
+    return ifelse(d < 0, zero(d), d)
+end
+
+function _correct_distance_to_set(d::AbstractVector{T}) where {T <: Real}
+    return [ifelse(di < 0, zero(T), di) for di in d]
+end
+
+function distance_to_set(v, s)
+    length(v) != MOI.dimension(s) && throw(DimensionMismatch("Mismatch between v and s"))
+    d = _distance(v, s)
+    return _correct_distance_to_set(d)
+end
+
+_distance(v::Real, s::MOI.LessThan) = v - s.upper
+_distance(v::Real, s::MOI.GreaterThan) = s.lower - v
+_distance(v::Real, s::MOI.EqualTo) = abs(v - s.value)
+
+_distance(v::Real, s::MOI.Interval) = max(s.lower - v, v - s.upper)
+
+_distance(v::AbstractVector{<:Real}, ::MOI.Reals) = false * v
```

Comment:
Why not simply `zero.(v)` ?

Reply:
I'll replace with `zeros(length(v))`, I'm not sure a broadcast would be eliminated